src/routes/zome_call.rs

```diff
@@ -0,0 +1,65 @@
+use axum::extract::{FromRequestParts, Path, Query};
+use base64::{prelude::BASE64_URL_SAFE, Engine};
+use holochain_types::dna::DnaHash;
+use serde::Deserialize;
+
+use crate::{HcHttpGatewayError, HcHttpGatewayResult};
+
+#[derive(Debug, Deserialize)]
+#[allow(unused, reason = "Temporarily unused fields")]
+pub struct ZomeCallParams {
+    dna_hash: DnaHash,
+    coordinator_identifier: String,
+    zome_name: String,
+    function_name: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct RawZomeCallParams {
+    dna_hash: String,
+    coordinator_identifier: String,
+    zome_name: String,
+    function_name: String,
+}
+
+impl<S> FromRequestParts<S> for ZomeCallParams
+where
+    S: Send + Sync,
+{
+    type Rejection = HcHttpGatewayError;
+
+    async fn from_request_parts(
+        parts: &mut axum::http::request::Parts,
+        state: &S,
+    ) -> Result<Self, Self::Rejection> {
+        let Path(raw_params) = Path::<RawZomeCallParams>::from_request_parts(parts, state).await?;
+        let dna_hash = DnaHash::try_from(raw_params.dna_hash)?;
+
+        Ok(ZomeCallParams {
+            dna_hash,
+            coordinator_identifier: raw_params.coordinator_identifier,
+            zome_name: raw_params.zome_name,
+            function_name: raw_params.function_name,
+        })
+    }
+}
```

Comment:
The default extractor was not producing invalid length errors when trying to convert the dna hash query param to a `DnaHash` type. Invalid DNA hashes now produce `400` error responses

https://github.com/holochain/hc-http-gw/pull/14#discussion_r1981349232